fcrepo-kernel-modeshape/src/test/java/org/fcrepo/integration/kernel/modeshape/observer/SimpleObserverIT.java

```diff
@@ -116,30 +125,94 @@ public void contentEventCollapsing() throws RepositoryException, InvalidChecksum
         se.logout();
 
-        try {
-            sleep(500);
-        } catch (final InterruptedException e) {
-            e.printStackTrace();
-        }
+        awaitEvent("/object3", NODE_ADDED, REPOSITORY_NAMESPACE + "lastModified");
 
         assertEquals("Node and content events not collapsed!", (Integer) 1, eventBusMessageCount);
+    }
+
+    @Test(timeout = TIMEOUT)
+    public void TestMoveEvent() throws RepositoryException {
+
+        final Session se = repository.login();
+        final NodeService ns = new NodeServiceImpl();
+
+        final Node n = se.getRootNode().addNode("/object4");
+        n.addMixin(FEDORA_CONTAINER);
+        n.addNode("/child1").addMixin(FEDORA_CONTAINER);
+        n.addNode("/child2").addMixin(FEDORA_CONTAINER);
+        se.save();
+        ns.moveObject(se, "/object4", "/object5");
+        se.save();
+        se.logout();
 
+        awaitEvent("/object4", NODE_ADDED);
+        awaitEvent("/object4/child1", NODE_ADDED);
+        awaitEvent("/object4/child2", NODE_ADDED);
+        awaitEvent("/object5", NODE_MOVED);
+        awaitEvent("/object5/child1", NODE_MOVED);
+        awaitEvent("/object5/child2", NODE_MOVED);
+        awaitEvent("/object4", NODE_REMOVED);
+        awaitEvent("/object4/child1", NODE_REMOVED);
+        awaitEvent("/object4/child2", NODE_REMOVED);
+
+        assertEquals("Move operation didn't generate additional events", (Integer) 9, eventBusMessageCount);
+    }
+
+    @Test(timeout = TIMEOUT)
+    public void TestMoveContainedEvent() throws RepositoryException {
+
+        final Session se = repository.login();
+        final NodeService ns = new NodeServiceImpl();
+
+        final Node n = se.getRootNode().addNode("/object6");
+        n.addMixin(FEDORA_CONTAINER);
+        final Node child = n.addNode("/object7");
+        child.addMixin(FEDORA_CONTAINER);
+        child.addNode("/child1").addMixin(FEDORA_CONTAINER);
+        child.addNode("/child2").addMixin(FEDORA_CONTAINER);
+        se.save();
+        ns.moveObject(se, "/object6/object7", "/object6/object8");
+        se.save();
+        se.logout();
+
+        awaitEvent("/object6", NODE_ADDED);
+        awaitEvent("/object6/object7", NODE_ADDED);
+        awaitEvent("/object6/object7/child1", NODE_ADDED);
+        awaitEvent("/object6/object7/child2", NODE_ADDED);
+        awaitEvent("/object6/object8", NODE_MOVED);
+        awaitEvent("/object6/object8/child1", NODE_MOVED);
+        awaitEvent("/object6/object8/child2", NODE_MOVED);
+        awaitEvent("/object6/object7", NODE_REMOVED);
+        awaitEvent("/object6/object7/child1", NODE_REMOVED);
+        awaitEvent("/object6/object7/child2", NODE_REMOVED);
+        // should produce two of these
+        awaitEvent("/object6", PROPERTY_CHANGED);
+
+        assertEquals("Move operation didn't generate additional events", (Integer) 12, eventBusMessageCount);
+    }
+
+    private void awaitEvent(final String id, final EventType eventType, final String property) {
```

Comment:
Not additionally, more that you could get rid of the multiple timeouts, and if someone wants to add a test later, they wouldn't have to remember to add a timeout themselves. Not a blocker-- your call.